src/test/java/de/mediathekview/mserver/crawler/wdr/parser/WdrFilmDeserializerTest.java

```diff
@@ -0,0 +1,273 @@
+    final String aJsUrl,
+    final String aJsFile,
+    final String aM3u8UrlVideo,
+    final String aM3u8FileVideo,
+    final String aM3u8UrlSignLanguage,
+    final String aM3u8FileSignLanguage,
+    final String aM3u8UrlAudioDescription,
+    final String aM3u8FileAudioDescription,
+    final String aTopic,
+    final String aExpectedTitle,
+    final LocalDateTime aExpectedDate,
+    final Duration aExpectedDuration,
+    final String aExpectedDescription,
+    final String aExpectedSubtitle,
+    final String aExpectedUrlSmall,
+    final String aExpectedUrlNormal,
+    final String aExpectedUrlHd,
+    final String aExpectedUrlSignLanguageSmall,
+    final String aExpectedUrlSignLanguageNormal,
+    final String aExpectedUrlSignLanguageHd,
+    final String aExpectedUrlAudioDescriptionSmall,
+    final String aExpectedUrlAudioDescriptionNormal,
+    final String aExpectedUrlAudioDescriptionHd,
+    final GeoLocations[] aExpectedGeoLocations) {
+    requestUrl = aRequestUrl;
+    filmPageFile = aFilmPageFile;
+    jsUrl = aJsUrl;
+    jsFile = aJsFile;
+    m3u8UrlVideo = aM3u8UrlVideo;
+    m3u8FileVideo = aM3u8FileVideo;
+    m3u8UrlSignLanguage = aM3u8UrlSignLanguage;
+    m3u8FileSignLanguage = aM3u8FileSignLanguage;
+    m3u8UrlAudioDescription = aM3u8UrlAudioDescription;
+    m3u8FileAudioDescription = aM3u8FileAudioDescription;
+    topic = aTopic;
+    expectedTitle = aExpectedTitle;
+    expectedDate = aExpectedDate;
+    expectedDuration = aExpectedDuration;
+    expectedDescription = aExpectedDescription;
+    expectedSubtitle = aExpectedSubtitle;
+    expectedUrlSmall = aExpectedUrlSmall;
+    expectedUrlNormal = aExpectedUrlNormal;
+    expectedUrlHd = aExpectedUrlHd;
+    expectedUrlSignLanguageSmall = aExpectedUrlSignLanguageSmall;
+    expectedUrlSignLanguageNormal = aExpectedUrlSignLanguageNormal;
+    expectedUrlSignLanguageHd = aExpectedUrlSignLanguageHd;
+    expectedUrlAudioDescriptionSmall = aExpectedUrlAudioDescriptionSmall;
+    expectedUrlAudioDescriptionNormal = aExpectedUrlAudioDescriptionNormal;
+    expectedUrlAudioDescriptionHd = aExpectedUrlAudioDescriptionHd;
+    expectedGeoLocations = aExpectedGeoLocations;
+  }
+  
+  @Test
+  public void test() throws IOException {
+    final Document document = Jsoup.parse(FileReader.readFile(filmPageFile));
+    setupSuccessfulResponse(jsUrl, jsFile);
+    if (!m3u8UrlVideo.isEmpty()) {
+      setupSuccessfulResponse(m3u8UrlVideo, m3u8FileVideo);
+    }
+    if (!m3u8UrlSignLanguage.isEmpty()) {
+      setupSuccessfulResponse(m3u8UrlSignLanguage, m3u8FileSignLanguage);
+    }
+    if (!m3u8UrlAudioDescription.isEmpty()) {
+      setupSuccessfulResponse(m3u8UrlAudioDescription, m3u8FileAudioDescription);
+    }
+    
+    Optional<Film> actual = new WdrFilmDeserializer("http:", Sender.WDR)
+      .deserialize(new TopicUrlDTO(topic, requestUrl), document);
+    
+    assertThat(actual.isPresent(), equalTo(true));
+    
+    Film actualFilm = actual.get();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Call "actual.isPresent()" before accessing the value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3655)